tools/e2e-commons/plugins/e2e-plan-data-interceptor.php

```diff
@@ -29,6 +29,72 @@ function e2e_intercept_plan_data_request( $return, $r, $url ) {
 		return $return;
 	}
 
+	if ( 1 === preg_match( sprintf( '/\/sites\/%d\/waf-rules/', $site_id ), $url ) ) {
+
+		$rules = <<<'RULES'
```

Comment:
It doesn't look like an easily maintainable approach. Is there anything else we can do here? 

Also, it looks like super specific for the WAF test, so maybe we can isolate it somehow?

Reply:
The intent of the test is to check that it "still blocks at all", so we only really need that one rule. Also this is our workaround for not creating an actual subscription.

We thought about moving that to a different file and register it on the same filter, but felt like it was technically also "data to do with the plan", so we left it in here. Would you think moving it to a different file would improve it?

Reply:
The "problem" here is that this plugin is part of a shared E2E infra and was created to help with plan mocking. Your change is quite specific to Jetpack plugin. It wouldn't break tests for other plugins, but it just messes up the separation of context. Ideally, this added logic should live in Jetpack E2Es folder, but since we don't have the logic to load functional plugins from within project folders, it should be good for now. @adimoldovan, what do you think?

Reply:
I would move this into a different file to be honest. 
It will also make things easier for when we'll have the logic to load functional plugins from within project folders (hopefully soon enough)

Reply:
Moved it to a separate file 🙂 Thanks for your input!